test/tests.cpp

```diff
@@ -3460,6 +3482,18 @@ TEST_CASE_METHOD(QInterfaceTestFixture, "test_forcem")
 
     REQUIRE(qftReg->ProbMask(0x7, 0x2) > 0.99);
     REQUIRE_FLOAT(qftReg->ProbMask(0xF, 0x2), 0.5);
+
+    qftReg->SetPermutation(0x0);
+    qftReg->H(1);
+    qftReg->CNOT(1, 2);
+    qftReg->H(3);
+    qftReg->CNOT(3, 4);
+    qftReg->H(5);
+    qftReg->CNOT(5, 6);
+    qftReg->CNOT(5, 7);
+
+    qftReg->ForceMReg(2, 5, 0x19, true);
```

Comment:
Is it worth adding a variation on this test that inverts the order of these two such that accidental side effects of ForceMReg can be caught?

Reply:
Now that you mention it, this part was added to the test exactly to catch the potential accidental side-effects I guessed the new optimized method could have, and this test _did_ catch a logical error. Incidentally, the error was _not_ in the new optimized method, but in the failure of `QUnit` to anticipate the "non-local correlation" between a single bit measured on the "wave function collapse" of the other potentially entangled bits, (which are specifically just those that share a shard `QEngine`).

That error was corrected. No, I don't think rearranging the gate order, in this case, tests anything fundamentally new, or covers more lines of code.

However, on a general note, I'm working on getting the Qrack Qiskit provider to pass all of Qiskit's unit tests. We have some failures, at the moment. I'm balancing that aspiration, to pass all of Qiskit's tests, with the need to produce a report for our Unitary Fund grant. Both will ultimately be achieved, as far I'm concerned. Most of the failures on those tests are due to gates and options we don't yet support, but I'll check the differential in failures between `QEngineOCL` or `QEngineCPU`, (assumed to be a "control case," as relatively simply implementations,) versus `QUnit`, (because it's this potential differential I'm worried about). We do seem to pass all of Pennylane's tests except general 3 qubit gates, further up that stack, simply because we don't provide general 3 qubit unitary gates, and Qiskit doesn't yet provide their decomposition, either.